default/python/AI/AIFleetMission.py

```diff
@@ -660,3 +725,85 @@ def __str__(self):
                                                                             (fleet and len(fleet.shipIDs)) or 0,
                                                                             CombatRatingsAI.get_fleet_rating(fleet_id),
                                                                             self.target or 'no target')
+
+    def _get_target_for_protection_mission(self):
+        """Get a target for a PROTECT_REGION mission.
+
+        1) If primary target (system target of this mission) is under attack, move to primary target.
```

Comment:
what if not strong enough?

Reply:
As is the case with current mission types, it will finally be handled by movement check code at later stage where more information is available.